mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb

```diff
@@ -1054,13 +1049,21 @@ func TestFrameworkProvider_LoadAndValidateFramework_impersonateServiceAccount(t
 			ExpectedDataModelValue: types.StringNull(),
 		},
 		// Handling empty strings in config
-		// TODO(SarahFrench) make these tests pass to address: https://github.com/hashicorp/terraform-provider-google/issues/14255
-		// "when impersonate_service_account is set as an empty array the field is treated as if it's unset, without error": {
-		// 	ConfigValues: fwmodels.ProviderModel{
-		// 		ImpersonateServiceAccount: types.StringValue(""),
-		// 	},
-		// 	ExpectedDataModelValue: types.StringNull(),
-		// },
+		"when impersonate_service_account is set as an empty string the field is treated as if it's unset, without error": {
+			ConfigValues: fwmodels.ProviderModel{
+				ImpersonateServiceAccount: types.StringValue(""),
+			},
+			ExpectedDataModelValue: types.StringNull(),
+		},
+		"when impersonate_service_account is set as an empty string in the config, an environment variable is used": {
+			ConfigValues: fwmodels.ProviderModel{
+				ImpersonateServiceAccount: types.StringValue(""),
+			},
+			EnvVariables: map[string]string{
+				"GOOGLE_IMPERSONATE_SERVICE_ACCOUNT": "value-from-env@example.com",
+			},
+			ExpectedDataModelValue: types.StringValue("value-from-env@example.com"),
+		},
```

Comment:
This is a test case that was missing, I've also added an equivalent in `.../provider/provider_internal_test.go` in this PR